pkg/reconciler/reconciler.go

```diff
@@ -95,6 +96,16 @@ type Base struct {
 	// performance benefits, raw logger also preserves type-safety at
 	// the expense of slightly greater verbosity.
 	Logger *zap.SugaredLogger
+
+	// watchers that can be stopped via Stop().
+	watchers []watch.Interface
+}
+
+// Stop stops all the watchers that reconciler has.
+func (b *Base) Stop() {
+	for _, w := range b.watchers {
+		w.Stop()
+	}
```

Comment:
I don't like this. :-/

Instead of this (or the new state) can we simply start a go routine that listens for the stop channel in `opt Options` to close and closes these?